### PR TITLE
Glm reports return HTMLReports

### PR DESCRIPTION
**Pull request by NicolasGensollen**
Addresses #2602  .

Changes proposed in this pull request:

- `make_glm_report` returns an `HTMLReport` rather than an `HTMLDocument`
- Split head and body templates for `glm` reports

Having the `glm` reports as `HTMLDocument` was causing an embeding issue in Jupyter Notebook because of the iframe wrapper implemented in `_repr_html` (see #2602). 

One way to test this is to open a Notebook and run the following code:

```python
import numpy as np
import pandas as pd
from nilearn import datasets
haxby_dataset = datasets.fetch_haxby()

# repetition has to be known
TR = 2.5

# Load target information as string and give a numerical identifier to each
behavioral = pd.read_csv(haxby_dataset.session_target[0], sep=' ')
conditions = behavioral['labels'].values

# Record these as an array of sessions
sessions = behavioral['chunks'].values
unique_sessions = behavioral['chunks'].unique()

# fMRI data: a unique file for each session
func_filename = haxby_dataset.func[0]

events = {}
# events will take  the form of a dictionary of Dataframes, one per session
for session in unique_sessions:
    # get the condition label per session
    conditions_session = conditions[sessions == session]
    # get the number of scans per session, then the corresponding
    # vector of frame times
    n_scans = len(conditions_session)
    frame_times = TR * np.arange(n_scans)
    # each event last the full TR
    duration = TR * np.ones(n_scans)
    # Define the events object
    events_ = pd.DataFrame(
        {'onset': frame_times, 'trial_type': conditions_session, 'duration': duration})
    # remove the rest condition and insert into the dictionary
    events[session] = events_[events_.trial_type != 'rest']
    
z_maps = []
conditions_label = []
session_label = []

# Instantiate the glm
from nilearn.glm.first_level import FirstLevelModel
glm = FirstLevelModel(t_r=TR,
                      mask_img=haxby_dataset.mask,
                      high_pass=.008,
                      smoothing_fwhm=4,
                      memory='nilearn_cache')

events[session].trial_type.unique()
from nilearn.image import index_img
for session in unique_sessions:
    # grab the fmri data for that particular session
    fmri_session = index_img(func_filename, sessions == session)

    # fit the glm
    glm.fit(fmri_session, events=events[session])

    # set up contrasts: one per condition
    conditions = events[session].trial_type.unique()
    for condition_ in conditions:
        z_maps.append(glm.compute_contrast(condition_))
        conditions_label.append(condition_)
        session_label.append(session)
        
from nilearn.image import mean_img
from nilearn.reporting import make_glm_report
mean_img_ = mean_img(func_filename)
report = make_glm_report(glm,
                         contrasts=conditions,
                         bg_img=mean_img_
                        )
report.save_as_html('test-glm-report.html')
report  # This report can be viewed in a notebook
```

### CODE REVIEW

**Review by bthirion — APPROVED** [review]
Thx for looking at that ! 
Hard for me to check all details but the result LGTM !

**Comment by bthirion** [comment]
Any other review ?

File: nilearn/reporting/html_report.py [hunk 1]
```diff
@@ -165,11 +165,11 @@ class HTMLReport(HTMLDocument):
     Methods such as save_as_html(), open_in_browser()
     are inherited from HTMLDocument
     """
-    def __init__(self, head_tpl, body):
+    def __init__(self, head_tpl, body, head_values={}):
```
Comment (jeromedockes) [inline nilearn/reporting/html_report.py:168]: detail: maybe `head_mapping` (the parameter to `safe_substitute` is "mapping") ? or something more explicit like `head_template_parameters`

**Review by jeromedockes — APPROVED** [review]
LGTM, thanks!

**Pull request merged**